hcloud/firewall.go

```diff
@@ -298,7 +298,35 @@ type FirewallSetRulesOpts struct {
 
 // SetRules sets the rules of a Firewall.
 func (c *FirewallClient) SetRules(ctx context.Context, firewall *Firewall, opts FirewallSetRulesOpts) ([]*Action, *Response, error) {
-	reqBody := firewallSetRulesOptsToSchema(opts)
+	s := firewallSetRulesOptsToSchema(opts)
+
+	// We can't use the FirewallRule struct here, because unlike in the response some fields must be omitted when empty.
+	type firewallRule struct {
```

Comment:
Should we make this a `schema.FirewallRuleRequest` instead to clearly separate the API schema code from business code?